macros/rp-connect-components.js

```diff
@@ -1,6 +1,117 @@
+
+    const table = document.getElementById('componentTable');
+    const trs = table.getElementsByTagName('tr');
+
+    for (let i = 1; i < trs.length; i++) {
+        const row = trs[i];
+        const nameTd = row.querySelector('td[id^="componentName-"]');
+        const typeTd = row.querySelector('td[id^="componentType-"]');
+        const supportTd = row.querySelector('td[id^="componentSupport-"]');
+        const typeDropdown = typeTd ? typeTd.querySelector('.type-dropdown') : null;
+
+        if (nameTd && typeTd) {
+            const nameText = nameTd.textContent.trim().toLowerCase();
+            const typeText = typeTd.textContent.trim().toLowerCase().split(', ').map(item => item.trim());
+            const supportText = supportTd ? supportTd.textContent.trim().toLowerCase() : '';
+
+            // Determine if the row should be shown
+            const showRow =
+                ((!nameInput || nameText.includes(nameInput)) &&
+                (typeFilter.length === 0 || typeFilter.some(value => typeText.includes(value))) &&
+                (!supportTd || supportFilter.length === 0 || supportFilter.some(value => supportText.includes(value)))
+                );
+
+            row.style.display = showRow ? '' : 'none';
+
+            if (showRow && typeFilter.length > 0 && typeDropdown) {
+                const matchingOption = Array.from(typeDropdown.options).find(option => 
+                    typeFilter.includes(option.text.toLowerCase())
+                );
+                if (matchingOption) {
+                    typeDropdown.value = matchingOption.value;
+                    updateComponentUrl(typeDropdown, false);
+                }
+            }
+        } else {
+            row.style.display = 'none'; // Hide row if essential cells are missing
+        }
+    }
+}
+
+  const capitalize = s => s && s[0].toUpperCase() + s.slice(1);
+
+  function processConnectors(parsedData) {
+    return parsedData.data.reduce((connectors, row) => {
+      const { connector, commercial_name, type, support_level, is_licensed, url } = row;
+      if (!connectors[connector]) {
+        connectors[connector] = {
+          types: new Map(),
+          supportLevels: new Map(),
+          isLicensed: is_licensed === 'y' ? 'Yes' : 'No',
+          urls: new Set()
+        };
+      }
+      connectors[connector].types.set(capitalize(type), url);
+      if (url) connectors[connector].urls.add(url);
+      if (!connectors[connector].supportLevels.has(support_level)) {
+        connectors[connector].supportLevels.set(support_level, new Set());
+      }
+      connectors[connector].supportLevels.get(support_level).add(commercial_name);
+      return connectors;
+    }, {});
+  }
+
+  function generateConnectorsHTMLTable(connectors, isCloud) {
+    return Object.entries(connectors).map(([connector, details], id) => {
+      const { types, supportLevels, isLicensed, urls } = details;
+      const firstUrl = urls.values().next().value || '#';
```

Comment:
When we switch to Ash's CSV, they'll be a time when new connectors are in the catalog but we don't have docs for them yet. In this case, it's better to omit the link rather than provide an empty one to avoid users thinking that the link is broken.